### PR TITLE
Fix SmartPlaylist Editor

### PR DESCRIPTION
**Pull request by Nine-H**
- refactored c style `sp` to `smart_playlist`
- added some margins
- minor cleaning to associate properties and objects

### CODE REVIEW

**Comment by Nine-H** [comment]
yeah nah, I'm completely at a loss as to why the remove button is not showing. I mentioned this to dan on slack. There's a couple more bugs I'm aiming to solve before I land this.

**Comment by Nine-H** [comment]
never mind, I reread my post and the inception music played in my head and thought "show, show, show, show, show(), `show()`, the show method! and sure enough the code was abusing it instead of calling one nice `show_all()`.

**Comment by Bat41** [comment]
Often, calling `show_all` after the initialization of the widgets often solves this kinds of problems (and it looks like it works here too).

File: src/Dialogs/SmartPlaylistEditor.vala [hunk 1]
```diff
@@ -191,263 +197,256 @@ public class Noise.SmartPlaylistEditor : Gtk.Dialog {
...
+        private Gtk.Button remove_button;
+        private Gtk.Entry value_entry;
+
+        private GLib.HashTable<int, SmartQuery.ComparatorType> comparators;
+
+        public signal void removed ();
+        public signal void changed ();
+
+        public SmartPlaylistEditorQuery (SmartQuery q) {
+            _q = q;
```
Comment (Bat41) [inline src/Dialogs/SmartPlaylistEditor.vala]: May be nice to use `Object` and `construct` here too.

**Comment by Nine-H** [comment]
currently broken due to some property access fuckery. need some sleep though and this looks better even if it doesn't work as good.

**Comment by Bat41** [comment]
I made a few changes, if they look good to you and you don't want to add anything here feel free to merge this branch.

**Comment by Nine-H** [comment]
your code looks good but there's some regressions around saving and loading.

**Comment by danrabbit** [comment]
This branch seems to have stalled and now conflicts with master. Closing for now. Feel free to reopen if work on this resumes :)

**Pull request closed without merging**